cobaya/likelihood.py

```diff
@@ -58,8 +58,7 @@ def current_logp(self) -> float:
         :return:  log likelihood from the current state as a scalar
         """
         value = self.current_state["logp"]
-        # Unfortunately, numpy arrays are not derived from Sequence, so the test is ugly
-        if hasattr(value, "__len__"):
+        if isinstance(value, Sized):
             value = value[0]
```

Comment:
Sized does not imply indexable, Sequence or MutableSequence?

Reply:
Note the Sized form is also nearly ten times slower than hasattr, which I think may be why we were using that

Reply:
Sized simply implies that it has length (as opposed to Sequence, which is not compatible with numpy arrays). Reverting where applicable. I though you preferred stronger type checks and less duck-typing, but fine to me.